### PR TITLE
prioritize config transit operator over gtfs agency

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
In tandem with https://github.com/opentripplanner/otp-react-redux/pull/734, this prioritizes using the Transit Operators listed in the configurations file over the operators in the GTFS route.

### CODE REVIEW

**Comment by demory** [comment]
@amy-corson-ibigroup I think I figured out why I was having problems testing this alongside otp-rr -- the dev branch is off a very old copy of the otp-ui code, and the file you edited (`itinerary-body/src/index.js`) no longer exists in otp-ui (the module has since been typescriptified).

When I tried to copy this into my otp-rr setup I was getting very odd errors like it not being able to find moment and other old dependencies.

Are you able to fast-forward the branch and migrate your edits to the TS code?

**Comment by demory** [comment]
Looks like the operator name isn't making it into the alt text:
image

File: packages/itinerary-body/src/TransitLegBody/index.tsx [hunk 1]
```diff
@@ -214,7 +222,7 @@ class TransitLegBody extends Component<Props, State> {
                               id: "otpUi.TransitLegBody.agencyLogo"
                             },
                             {
-                              agencyName
+                              transitOperatorName
```
[the remark this answers is not in the document]
Comment (amy-corson-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx]: Sorry about that! Fixed in 06e599fe838792c08763a5f259ab284474af5599

**Comment by demory** [comment]
Now seeing some broken examples for the Itinerary body in storybook, can you confirm if you're seeing as well:

image

**Review by demory — APPROVED** [review]

File: packages/itinerary-body/src/TransitLegBody/index.tsx [hunk 2]
```diff
@@ -157,6 +157,13 @@ class TransitLegBody extends Component<Props, State> {
       leg
     );
 
+    // If the config contains an operator name, prefer that one over the
+    // one provided by OTP
+
+    const transitOperatorName = transitOperator?.name
+      ? transitOperator.name
+      : agencyName;
+
```
Comment (miles-grant-ibigroup) [inline packages/itinerary-body/src/TransitLegBody/index.tsx:166]:
```suggestion
    const transitOperatorName = transitOperator?.name || agencyName;
```

Does this work?

**Review by miles-grant-ibigroup — APPROVED** [review]
Wow was the prop we needed there but not being used! Well done on getting the fix to be this simple! Just one small style nitpick.

File: packages/itinerary-body/src/TransitLegBody/index.tsx (hunk 2 above)
Comment (amy-corson-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx]: Yes I think so! Thank you!

**Pull request merged**